### PR TITLE
dotnet tool uninstall should find the first manifest file contain package id

### PR DESCRIPTION
**Pull request by wli3**
fix https://github.com/dotnet/cli/issues/11139  , a lot of file touched. But most of them are locs

Instead of the first manifest file regardless. So, the behavior is aligned with tool update.
Extract method ToolManifestFinderExtensions from tool update. And use them in both places.

I changed the following to make it generic. Since the user just type the command, I don't think uninstall or update in the error message is significant.

"If you intended **to uninstall** on a global tool, use the `--global` option for the command." to "If you intended **to perform an action** on a global tool, use the `--global` option for the command."

### CODE REVIEW

File: src/dotnet/commands/dotnet-tool/uninstall/LocalizableStrings.resx [hunk 1]
```diff
@@ -168,7 +168,7 @@
   <data name="UninstallLocalToolSucceeded" xml:space="preserve">
     <value>Tool '{0}' was successfully uninstalled and removed from manifest file {1}.</value>
   </data>
-  <data name="NoManifestGuide" xml:space="preserve">
-    <value>If you intended to uninstall a global tool, add `--global` to the command.</value>
+  <data name="NoManifestFileContainPackageId" xml:space="preserve">
+    <value>Cannot find a manifest file contain the package id '{0}'.</value>
```
Comment (peterhuene) [inline src/dotnet/commands/dotnet-tool/uninstall/LocalizableStrings.resx]:
```suggestion
    <value>Cannot find a manifest file that contains package id '{0}'.</value>
```

**Comment by peterhuene** [comment]
Looks like CI failed with the SIGINT test.  I'll look into that again.  I need to figure out why the fix for the race condition didn't seem to have the desired effect (it certainly has made it less "racy", though, since we see it less often).

**Comment by peterhuene** [comment]
I'll put up a PR to disable the test for now and reopen that tracking issue you had and assign it to me.

**Comment by wli3** [comment]
Thanks!

**Pull request merged**